### PR TITLE
lua: introduce interface for prbuf

### PR DESCRIPTION
**Pull request by Korablev77**
Subj.

### CODE REVIEW

File: changelogs/unreleased/prbuf.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## feature/lua
+
+* Introduced new buffer - partitioned ring buffer, which is available in Lua
+  via require('buffer').prbuf_create() and require('buffer').prbuf_open().
```
Comment (locker) [inline changelogs/unreleased/prbuf.md:4]: I'm not sure we should make it public. Moreover, I don't even see how we could use it internally. Why do you need it?

File: test/app-tap/buffer.test.lua [hunk 2]
```diff
@@ -52,8 +52,54 @@ local function test_cord_ibuf(test)
     cord_ibuf_put(ibuf2)
 end
 
+local function fill_memory(memory, size)
```
Comment (locker) [inline test/app-tap/buffer.test.lua:55]: We don't use TAP for new tests. Only Luatest.

File: changelogs/unreleased/prbuf.md (hunk 1 above)
Comment (locker) [reply changelogs/unreleased/prbuf.md:4]: Okay. It'd be nice to mention in the commit message. Still, I don't think we need to document it or mention it in the changelog, at least now, when nobody asked for it. We might want to change the prbuf API in the future. Making the API public would render such changes complicated.

Comment (Korablev77) [reply changelogs/unreleased/prbuf.md]: Flight recorder is an enterprise feature, so that I've skipped mentioning it in the commit message.

**Comment by Korablev77** [comment]
Rewrite test using luatest; removed changelog & docbot request.

**Review by locker — APPROVED** [review]

**Pull request merged**